app/src/main/java/com/sdp13epfl2021/projmag/curriculumvitae/CurriculumVitae.kt

```diff
@@ -2,139 +2,168 @@
 
 package com.sdp13epfl2021.projmag.curriculumvitae
 
-import android.net.Uri
 import android.os.Parcelable
 import kotlinx.parcelize.Parcelize
 import java.io.Serializable
 
 
+@Parcelize
 /**
- * A curriculum vitae
+ * This a representation of a Curriculum Vitae
+ *
+ * @property summary the summary of the CV
+ * @property education the list of educations the the person received
+ * @property jobExperience the list of jobs the person had, its professional experience
+ * @property languages the list of languages, along their respective levels
+ * @property skills various other skills
  */
-@Parcelize
 data class CurriculumVitae(
     val summary: String,
     val education: List<PeriodDescription>,
     val jobExperience: List<PeriodDescription>,
-    val languages: List<LanguageLevel>,
+    val languages: List<Language>,
     val skills: List<SkillDescription>,
 ) : Parcelable, Serializable {
 
-    var uri: Uri? = null
-        set(value) {
-            if (field == null) {
-                field = value
-            }
-        }
 
-    companion object {
-        interface Validate {
-            fun isValid(): Boolean
-        }
+    @Parcelize
+    /**
+     * Description a job/school over a period of time
+     *
+     * @property name the name of the occupation concerned by this period
+     * @property location the name of the location of this period of activity
+     * @property description a description of what has been done during this period
+     * @property from the beginning year
+     * @property to the ending year
+     */
+    data class PeriodDescription(
+        val name: String,
+        val location: String,
+        val description: String,
+        val from: Int,
+        val to: Int
+    ) : Validateable, Parcelable, Serializable {
 
         private val VALID_YEARS = 1900..2100
```

Comment:
Because VALID_YEARS is inside the PeriodDescription class now, when we try to serialize a CV, it failed because kotlin.ranges.IntRange is not serializable

Reply:
Using theses annotations seems to fix the problem
@IgnoredOnParcel
@Transient